src/handler/index.ts

```diff
@@ -35,45 +35,43 @@ export const handler = async (event: DynamoDBStreamEvent, context: Context): Pro
   const emailMessages: EmailMessageRequest[] = [];
   logger.info(`Received ${records.length} records from DynamoDB.`);
   records.forEach((record) => {
-    // Discard INSERT and REMOVE events
-    if (record.eventName !== 'MODIFY') {
-      logger.info(`Discarding ${record.eventName} event.`);
+    logger.debug(`Processing record: ${JSON.stringify(record)}`);
+
+    let message: AvailabilityChangeData;
+    try {
+      const body: Record<string, string> = <Record<string, string>> JSON.parse(record.body);
+      message = JSON.parse(body.Message) as AvailabilityChangeData;
+    } catch (parseError) {
+      logger.error(`Error while parsing message body for messageId: ${record.messageId}`);
+      logger.error(parseError);
       return;
     }
 
-    logger.debug(`Processing record: ${JSON.stringify(record)}`);
-
     // Extract data from event
     let availabilityData: AvailabilityChangeData;
     try {
-      availabilityData = extractAvailabilityData(record);
+      availabilityData = validateAvailabilityData(message);
     } catch (err2) {
       logger.error((err2 as Error).message);
       return; // skip to next record
     }
-    const {
-      oldAvailability,
-      newAvailability,
-    } = availabilityData;
+    const { availability } = availabilityData;
```

Comment:
should we expand the destructuring?

Reply:
I've got rid of the destrucing on availability as none of the others were getting deconstructed just to be set again into the object for `buildSQSMessage` to handle